internal/controller/storageclient_controller.go

```diff
@@ -421,9 +424,35 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	//Once the owner references are removed from resources, StorageClaim is deleted
+	if err = r.deleteStorageClaim(); err != nil {
+		return reconcile.Result{}, err
+	}
 	return reconcile.Result{}, nil
 }
 
+func (r *storageClientReconcile) deleteStorageClaim() error {
+	storageClaimList := &metav1.PartialObjectMetadataList{}
+	storageClaimList.SetGroupVersionKind(v1alpha1.GroupVersion.WithKind("StorageClaimList"))
+	if err := r.list(storageClaimList); err != nil {
+		return fmt.Errorf("Error listing StorageClaimList: %v", err)
+	}
+
+	for _, storageClaim := range storageClaimList.Items {
+		if err := r.Client.Delete(r.ctx, &storageClaim); err != nil {
+			return fmt.Errorf("Failed to delete StorageClaim %q: %v", storageClaim.Name, err)
+		}
+
+		original := &storageClaim
+		if controllerutil.RemoveFinalizer(&storageClaim, storageClaimFinalizer) {
+			if err := r.Client.Patch(r.ctx, &storageClaim, client.MergeFrom(original)); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
```

Comment:
We should not be deleting arbitrary storage claims, the side effect might be data lost for our customers. The 4.18 storage client owned 2 specific storage claims and our responsibility is to just clean up after these 2 specific ones